proposals/relaxed-simd/entropy.md

```diff
@@ -0,0 +1,57 @@
+* Compatibility risks, i.e. if code compiled for one browser works differently on a different browser (This is specific to observable behavior changes, and not performance differences)
+
+This document is an attempt to collate information that is already available in the issues filed for each of the operations on fingerprinting risk, quantify the risk for fingerprinting/compat issues and provide some information about usage patterns. 
+
+## Instruction summary
+
+* **relaxed i8x16.swizzle**<br>
+  *Entropy exposed:* Differences between Intel/ARM/RISC-V<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Low, as the differences exposed are for out of range indices<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/21)
+* **relaxed i32x4.trunc_{f32x4, f64x2} operations**<br>
+  *Entropy exposed:* Differences between Intel/ARM/RISC-V<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Low, as the differing behavior is for out of range values, and NaNs<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/21)<br>
+* **qfma, qfmns**<br>
+  *Entropy exposed:* Differences between hardware that has native FMA support, and hardware that does not.<br>
+  *Deterministic lowering:* Not available, but depending on underlying hardware, the results can only be fused, or unfused.<br>
+  *Compat risk:* TBD<br>
+  *Mitigating reasons to include:* Most modern hardware do have native FMA support, performance wins are significant, operations are difficult to emulate<br>
+  [*Issue link*](https://github.com/WebAssembly/simd/pull/79)<br>
+* **{i8x16, i16x8, i32x4, i64x2}.laneselect**<br>
+  *Entropy exposed:* Intel/ARM<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Medium, architectures vary on which bit is used for lane selection<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/17)<br>
+* **{f32x4, f64x2}.{min,max}**<br>
+  *Entropy exposed:* Intel/ARM<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Low, varying outputs when one of the inputs is NaN, or  +0, -0<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/33)<br>
+* **I16x8.relaxed_q15mulr_s**<br>
+  *Entropy exposed:* Intel/ARM<br>
+  *Deterministic lowering:* Available<br>
+  *Compat risk:* Low, different behaviors only in the overflow case<br>
+  [*Issue link*](https://github.com/WebAssembly/relaxed-simd/issues/40)<br>
+* **Dot product instructions**<br>
+  *Entropy exposed:* Intel/ARM, and whether the Dot product extension is supported in the most optimal codegen<br>
+  *Deterministic lowering:* Available, but only when not using the most optimal codegen<br>
+  *Compat risk:* Medium for architectures that support the Dot product extensions as they vary in saturating/wrapping behavior of intermediate results<br>   *Mitigating reasons to include:* TBD<br>
```

Comment:
Specifically for when dot product extensions are used, I remember @Maratyszcza mentioning a ballpark performance difference, are you able to add a section here for mitigating reasons to include here?

Reply:
Not sure I understand the question. Are you asking about performance difference of native dot production extension vs emulation sequence?

Reply:
Yeah, I'm not looking for exact experimental data, but a ballpark of the expected win of adding this instruction. Details about expected usage in algorithms, i.e. similar to the FMA discussion about correctness, are there algorithms that rely on the Dot product extension being natively supported and would the user have to rely on some form of detection?